src/pages/AddressInfoPage/index.tsx

```diff
@@ -57,127 +59,106 @@ type ParamTypes = {
   id: string
 }
 
-const TransactionInfoPage = () => {
+type FetchedEntity = 'balance' | 'txNumber' | 'assets' | 'transactions'
+
+type SingleStringArgFunctions<T> = {
+  [K in keyof T]: T[K] extends (arg: string) => unknown ? K : never
+}[keyof T]
+
+const AddressInfoPage = () => {
   const theme = useTheme()
   const { id } = useParams<ParamTypes>()
   const { client, setSnackbarMessage, networkType } = useGlobalContext()
+  const isAppVisible = usePageVisibility()
+  const pageNumber = usePageNumber()
 
   const [addressBalance, setAddressBalance] = useState<AddressBalance>()
   const [addressTransactionNumber, setAddressTransactionNumber] = useState<number>()
   const [addressAssets, setAddressAssets] = useState<AddressAssetsResult>()
   const [addressTransactions, setAddressTransactions] = useState<Transaction[]>()
+  const [addressMempoolTransactions, setAddressMempoolTransactions] = useState<MempoolTransaction[]>()
   const [addressLatestActivity, setAddressLatestActivity] = useState<number>()
+  const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
+  const [exportModalShown, setExportModalShown] = useState(false)
+
+  const knownPendingTxs = useRef<MempoolTransaction[]>([])
 
-  const [loadings, setLoadings] = useState({
+  const [loadings, setLoadings] = useState<{ [key in FetchedEntity]: boolean }>({
     balance: true,
     txNumber: true,
     assets: true,
     transactions: true
   })
 
-  const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
+  const fetchDataGeneric = useCallback(
+    async <T,>(
+      fetchedEntity: FetchedEntity,
+      setStateAction: (value: T | undefined) => void,
+      apiCallName?: SingleStringArgFunctions<ExplorerProvider['addresses']>,
+      customApiCall?: () => Promise<T>
```

Comment:
nit: How about merging these 2 into one, since based on the implementation, only one of the 2 is used, even in the case if both are being provided? There is then no need to pass `undefined` as the third argument.

```suggestion
      apiCall: SingleStringArgFunctions<ExplorerProvider['addresses']> | (() => Promise<T>)
```

see below for completion

Reply:
I like it! ❇️